kolena/_experimental/object_detection/utils.py

```diff
@@ -79,18 +94,28 @@ def _compute_threshold_curve(
             zero_division=0,
         )
 
-        precisions.append(precision)
-        recalls.append(recall)
+        # avoid curves with one x-value and two y-values
+        if recall in recalls:
+            idx = recalls.index(recall)
+            precisions[idx] = max(precisions[idx], precision)
+        else:
+            precisions.append(precision)
+            recalls.append(recall)
         f1s.append(f1)
 
-    # Omit curves with only one point
-    if len(f1s) < 2:
+    # omit curves with no points
+    if len(f1s) == 0:
         return None
 
     if curve_type == "f1":
-        return Curve(x=thresholds, y=f1s, label=curve_label)
+        return Curve(x=thresholds, y=f1s, label=curve_label) if len(f1s) >= 2 else None
     else:
-        return Curve(x=recalls, y=precisions, label=curve_label)
+        # add a point to start the PR curve on the vertical axis if needed
+        if 0.0 not in recalls:
+            minpos = recalls.index(min(recalls))
```

Comment:
This fails if `recalls` is `[]` -- is it guaranteed that there's at least one value?

Reply:
check for `recall` length on line 107